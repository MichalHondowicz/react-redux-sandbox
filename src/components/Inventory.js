import React from 'react';
import BookItem from './BookItem';
import {base} from "../base";

class Inventory extends React.Component {

    constructor() {
        super();
        this.state = {}
    }

    componentDidMount() {
        this.ref = base.syncState('bookstore/books', {
            context: this,
            state: 'books'
        })
    }

    componentWillUnmount() {
        base.removeBinding(this.ref)
    }

    render() {

        let bookList = <h4>No books in inventory</h4>;

        if (Array.isArray(this.state.books)) {
            bookList = this.state.books.filter((book) => book.availability = true)
                .map(book => {
                    return <BookItem key={book.name} book={book} addToOrder={this.props.addToOrder}/>
                });
        }
        return (
            <div className='inventory col-md-6'>
                <h2>Bookstore inventory</h2>
                {bookList}
            </div>
        )
    }
}

export default Inventory;