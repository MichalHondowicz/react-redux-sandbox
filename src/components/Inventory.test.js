import ReactDOM from "react-dom";
import Inventory from "./Inventory";
import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'

configure({adapter: new Adapter()});

describe('Inventory test', () => {
    it('Inventory renders with no problems ', () => {
        const div = document.createElement('div');
        const books = [];
        ReactDOM.render(<Inventory books={books}/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Inventory Snapshot ok', () => {
        const books = [];
        const wrapper = shallow(<Inventory books={books}/>);
        expect(wrapper).toMatchSnapshot();
    })
});