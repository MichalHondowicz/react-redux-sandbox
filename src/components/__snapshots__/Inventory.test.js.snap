// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inventory test Inventory Snapshot ok 1`] = `ShallowWrapper {}`;
